### DISCUSSION TITLE
How do I check If a session has been loaded from storage or has been created from the sign-up

### DISCUSSION THREAD

**Discussion by kibitheM** [supabase-community/supabase-kt#307] [Q&A] [answered]
I have a question I would really appreciate if anyone could help me out. Iam new to Supabase.
With GoTrue, how do I check if the user of the current session who is opening the app, is already signed in, or has been signed in after going through the sign-up screen.

Okey, in my app, in the main activity Iam collecting the sessionState as a flow and responding to the Not Authenticated state by showing the sign-up screen and the Authenticated state by showing the main screen. However as soon as a user is successfully signed in from the sign-up screen, the session state changes, and they get redirected to the main screen...this is not what I want, as I want to collect more user information to update their profile in the next screen after they have signed up, at the same time to log them in automatically to the main screen when they are already signed in. I Guese what Iam asking is, is there a way for me to check whether the authenticated state of the user has been loaded from a previous session or has been created from the sign-up screen. Thanks in advance.

**Comment by jan-tennert** [comment]
> I have a question I would really appreciate if anyone could help me out. Iam new to Supabase. With GoTrue, how do I check if the user of the current session who is opening the app, is already signed in, or has been signed in after going through the sign-up screen.
> 
> Okey, in my app, in the main activity Iam collecting the sessionState as a flow and responding to the Not Authenticated state by showing the sign-up screen and the Authenticated state by showing the main screen. However as soon as a user is successfully signed in from the sign-up screen, the session state changes, and they get redirected to the main screen...this is not what I want, as I want to collect more user information to update their profile in the next screen after they have signed up, at the same time to log them in automatically to the main screen when they are already signed in. I Guese what Iam asking is, is there a way for me to check whether the authenticated state of the user has been loaded from a previous session or has been created from the sign-up screen. Thanks in advance.

There is no way to distinguish between those scenarios. If I'm understanding your question correctly, you want to ask extra information from your user after signing up? (And possibly store this information in an extra postgres table?).

**Reply by kibitheM** [reply]
Thank you for your reply...Yes, that is pretty much what I want...but I don't intend to store the information in an extra Postgres table, rather to the userMetadata, but I want to do this before navigating them to the main screen, because I want to represent some part of their userMetadata in the main screen, but if they are coming back to the app then there's no need for this that's why the authentication check is at the beginning of the app/onCreate() (the coming back part works alright) ...

 I know this may be more of a coding question than necessarily relating to the library, but i would really appreciate some advise on how I should go about this.. You see, the issue is as soon as my user press that sign-Up button they get navigated to the main screen ( because my main activity is listening and racting to the sessionState change)  therefore, I cannot collect any more sign-up information than say maybe the email and maybe the name which are shown on the sign-up screen...I figured out that the only way I might achieve this is to first collect all their information before finally showing the sign-up button, but I feal it is more understandable if I could be able to sign them up first before I can ask them some more app related questions

**Reply by jan-tennert** [reply]
You can pass in the user data directly in the sign up, you don't have to wait after the user was created

**Reply by jan-tennert** [reply, marked as answer]
Also you can access the user when the session state changed via `sessionState.session.user` (if the state is Authenticated)

**Reply by kibitheM** [reply]
Yeah. That did it. Thank you very much.
